### PR TITLE
feat(icon): adding hover style if there is an onclick on the icon

### PR DESCRIPTION
**Pull request by stranzhay**
this will automatically put hover styling on all icons if there is an onClick event attached

### CODE REVIEW

**Comment by stranzhay** [comment]
still need to fix a failing test

File: packages/icon/src/Icon.js [hunk 1]
```diff
@@ -2,11 +2,18 @@ import React from 'react';
...
-    className={classNames('icon', `icon-${name}`, size && `icon-${size}`, color && `text-${color}`, className)}
+    className={classNames(
+      'icon',
+      `icon-${name}`,
+      size && `icon-${size}`,
+      color && `text-${color}`,
+      onClick && { cursor: 'pointer' },
+      className
+    )}
     {...rest}
```
Comment (LauRoxx) [inline packages/icon/src/Icon.js:19]: does onClick need to be explicitly passed now that it's pulled out of ...rest?

**Comment by jordan-a-young** [comment]
Can you add a test checking for this behavior?

**Comment by availity-bot-cicd** [comment]
🚀 Netlify deployed **condescending-leavitt-66e607** as draft

https://61561f505a302a00bf819e41--condescending-leavitt-66e607.netlify.app

**Comment by availity-bot-cicd** [comment]
🚀 Netlify deployed **condescending-leavitt-66e607** as draft

https://61561ff1cdad570b3f562a8a--condescending-leavitt-66e607.netlify.app

**Comment by availity-bot-cicd** [comment]
🚀 Netlify deployed **condescending-leavitt-66e607** as draft

https://6156216ec71cb80c6bc274ba--condescending-leavitt-66e607.netlify.app

**Comment by availity-bot-cicd** [comment]
🚀 Netlify deployed **condescending-leavitt-66e607** as draft

https://615621cf34f6ad0d4a143f28--condescending-leavitt-66e607.netlify.app

**Comment by stranzhay** [comment]
commit lint is messed up i will close this pr and make a new one tomorrow morning

**Pull request closed without merging**